### PR TITLE
Refine Feign manual client doc

### LINKED ISSUE #2584: Hint in documentation how manual feign client can use spring mvc annotations
Just like in #1834 a third party provider returns a dynamic url to us that we have to call using a Feign client. This client needs to be described using spring mvc annotations (shared with the provider part, actually [spring-cloud-cloudfoundry-service-broker](https://github.com/spring-cloud/spring-cloud-cloudfoundry-service-broker/blob/420f903e897af71cf7f6ec2d14478e8ef9c0a3ad/spring-cloud-open-service-broker/src/main/java/org/springframework/cloud/servicebroker/controller/ServiceInstanceController.java#L76-L85))

 Following, the [documentation](https://github.com/spring-cloud/spring-cloud-netflix/blob/085935071558668a77f4c625454de6f90fb72be3/docs/src/main/asciidoc/spring-cloud-netflix.adoc#creating-feign-clients-manually), I have built the client with the Feign.Builder.

However, both documentation and some samples such as https://github.com/ryanjbaxter/manual-feign-demo/blob/f211837799ffa908d0bf660843dba10a6ec5db41/src/main/java/com/example/FooClient.java#L23 imply that this approach requires using native feign annotations and not springmvc annotations.

It took me some time before I found out that I could reuse some of the FeignClientFactoryBean logic to preserve springMVC annotation support

https://github.com/spring-cloud/spring-cloud-netflix/blob/64ad2ba20c1fc2b5556cb94386fecd0f4f65a816/spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/feign/FeignClientFactoryBean.java#L93

I would suggest hinting in documentation users about getting inspiration FeignClientFactoryBean in the "7.3 Creating Feign Clients Manually" section.

If this can help other users, here is the resulting code in my project along with associated end to end integration test:
https://github.com/orange-cloudfoundry/cf-ops-automation-broker/blob/af2609b5fad79138b1bfcc2e40fe50ad7f8e2a27/cf-ops-automation-broker-core/src/main/java/com/orange/oss/cloudfoundry/broker/opsautomation/ondemandbroker/osbclient/OsbClientFactory.java
https://github.com/orange-cloudfoundry/cf-ops-automation-broker/blob/af2609b5fad79138b1bfcc2e40fe50ad7f8e2a27/cf-ops-automation-cassandra-broker/src/test/java/com/orange/oss/cloudfoundry/broker/opsautomation/ondemandbroker/osbclient/OsbClientTestApplicationTest.java#L43-L48

ps: also cross-referencing related issues #1458 and #990 as to help discovery.

### PR DESCRIPTION
**Pull request by gberche-orange**
Hint about support of springmvc annotations in the manual client instanciation
Fixes #2584

### CODE REVIEW

**Comment by ryanjbaxter** [comment]
Maybe add some words about the `Contract` class and that you can pass `SpringMvcContract` to support the Spring MVC annotations.

**Comment by gberche-orange** [comment]
@ryanjbaxter I added more context about Feign [Contract](https://github.com/OpenFeign/feign/blob/a029afd8666b9a8ddeaecd1599e10b7b55f43bb4/core/src/main/java/feign/Contract.java#L34) into https://github.com/spring-cloud/spring-cloud-netflix/pull/2627/commits/994134ea35e9d30c48a86f35bcf34003b9cdc111

File: docs/src/main/asciidoc/spring-cloud-netflix.adoc [hunk 1]
```diff
@@ -1212,6 +1214,10 @@ provided by Spring Cloud Netflix.
 
 NOTE: `PROD-SVC` is the name of the service the Clients will be making requests to.
 
+NOTE: The Feign Contract object defines what annotations and values are valid on interfaces. The
```
Comment (ryanjbaxter) [inline docs/src/main/asciidoc/spring-cloud-netflix.adoc]: Use single ticks (\`) to highlight code (ie `Contract`)

**Comment by gberche-orange** [comment]
@ryanjbaxter formatting added into https://github.com/spring-cloud/spring-cloud-netflix/pull/2627/commits/8462745c9b7a1d88133a7e17ad70182e10cb6bd5

**Pull request merged**